product-service/swagger.yml

```diff
@@ -1,97 +1,194 @@
+                      data:
+                        type: object
+                        items:
+                          $ref: '#/components/schemas/Product'
+        '400':
+          content:
+            application/json:
+              schema:
+                type: string
+          description: Product not found
+        '500':
+          content:
+            application/json:
+              schema:
+                type: string
+          description: Something went wrong
+    options:
+      responses:
+        '200':
+          description: 200 response
+          headers:
+            Access-Control-Allow-Origin:
+              schema:
+                type: string
+            Access-Control-Allow-Methods:
+              schema:
+                type: string
+            Access-Control-Allow-Headers:
+              schema:
+                type: string
+          content: {}
+  /products:
     get:
-      tags:
-      - "products"
-      summary: "Find product by ID"
-      description: "Returns a single product"
-      operationId: "getProductById"
-      produces:
-      - "application/json"
-      parameters:
-      - name: "productId"
-        in: "path"
-        description: "ID of product to return"
-        required: true
-        type: "string"
+      description: Get all products
       responses:
-        "200":
-          description: "successful operation"
-          schema:
-            $ref: "#/definitions/Product"
-        "400":
-          description: "Invalid ID supplied"
-          
-definitions:
-  Product:
-    type: "object"
-    properties:
-      price:
-        type: "integer"
-        format: "int64"
-      count:
-        type: "integer"
-        format: "int64"
-      id:
-        type: "string"
-        format: "guid"
-      title:
-        type: "string"
-        description: "product title"
-      imageName:
-        type: "string"
-        description: "image name"
-      description:
-        type: "string"
-        description: "product description" 
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  code: 
+                    type: string
+                    enum:
+                      - "success"
+                      - "error"
+                  result:
+                    type: object
+                    properties:
+                      items:
+                        type: array
+                        items:
+                          $ref: '#/components/schemas/Product'
+                      
+        '500':
+          content:
+            application/json:
+              schema:
+                type: string
+          description: Internal server error
+    post: 
+      description: Create product
+      requestBody:
+          required: true
+          content:
+            application/x-www-form-urlencoded:

```

Comment:
A compelling approach! But why?

Reply:
sorry, I'm confusing a little bit, you about `required: true` or content: `application/x-ww..` ?

Reply:
Probably, there are three chairs:

if you about moving from 2.0 to 3.1 OPEN API: due to enjoyable automated code-generation from AWS API Gateway (stages section)

if you about `required: true`: sure! `required` only some fields (not the whole model), fix it later

if about replace `application/json` -> `application/x-www-urlendecoded` due to CORS issues

`(copy from MSDN)`
```
The only allowed values for the Content-Type header are (when POST):

application/x-www-form-urlencoded
multipart/form-data
text/plain
```

Thank you for your review!

Reply:
I have no idea what MSDN or frontend have to do with this... whatever... if it works for you, it is fine with me!

Reply:
Ok, could you say what is moment call your confusion in these changes? Maybe I could improve this 

I mean that I can't send cross-domain POST request by application/JSON header, only within  x-www-form-urlendecode` because CORS rules have limits

Fragment about CORS spec here: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS`

sorry, maybe I don't understand you correctly